File: express-backend/src/services/provider/ImdbMovieSearchProvider.ts
import { ExternalMovieSearchResult, ProviderSearchResult } from '../../dto/external-search';
import { MovieSearchProvider } from '../provider/MovieSearchProvider';

// IMDb 直连 Provider — 占位，与 Java 端 ImdbMovieSearchProvider 对齐
export class ImdbMovieSearchProvider implements MovieSearchProvider {
  id(): string {
    return 'imdb';
  }

  async search(query: string, page: number): Promise<ProviderSearchResult<ExternalMovieSearchResult>> {
    return {
      provider: this.id(),
      enabled: false,
      message: 'IMDb API 需要 AWS Data Exchange 签名调用，当前未实现，建议使用 OMDb（provider=omdb）',
      page,
      totalPages: 0,
      totalResults: 0,
      results: [],
    };
  }
}

File: legacy/java-backend/src/main/java/com/pixelreel/service/provider/ImdbMovieSearchProvider.java
package com.pixelreel.service.provider;

import com.pixelreel.config.ImdbProperties;
import com.pixelreel.dto.external.ExternalMovieSearchResult;
import com.pixelreel.dto.external.ProviderSearchResult;
import com.pixelreel.service.search.MovieSearchProvider;
import java.util.Collections;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ImdbMovieSearchProvider implements MovieSearchProvider {
  private final ImdbProperties imdbProperties;

  @Override
  public String id() {
    return "imdb";
  }

  @Override
  public ProviderSearchResult<ExternalMovieSearchResult> search(String query, int page) {
    ProviderSearchResult<ExternalMovieSearchResult> result = new ProviderSearchResult<>();
    result.setProvider(id());

    if (!imdbProperties.enabled()) {
      result.setEnabled(false);
      result.setMessage("IMDb 直连未启用，建议使用 OMDb（provider=omdb）");
      result.setPage(page);
      result.setTotalPages(0);
      result.setTotalResults(0);
      result.setResults(Collections.emptyList());
      return result;
    }

    result.setEnabled(false);
    result.setMessage("IMDb API 需要 AWS Data Exchange 签名调用，当前未实现");
    result.setPage(page);
    result.setTotalPages(0);
    result.setTotalResults(0);
    result.setResults(Collections.emptyList());
    return result;
  }
}


